docs/markdown/Machine-files.md

```diff
@@ -155,6 +153,14 @@ case). For compilers like MSVC and Clang-Cl, this is the path to a
 linker for Meson to invoke, such as `link.exe` or `lld-link.exe`.
 Support for `ld` is *new in 0.53.0*
 
+These values must be either strings or an array of strings. If a path
+contains spaces, it must be wrapped into an array or with double quotes,
+otherwise the first part will be taken as a path to the `ccache` executable:
```

Comment:
The second sentence is a run-on. Could you also add an example using double quotes? Does single quotes not work?

Reply:
Double quoting is considered a parser failure:

```
❯ msvc-dev-cmd --vsversion 2022 --arch amd64 -- meson setup build --native-file=./native-file.txt --wipe

ERROR: Malformed value in machine file variable 'cc'.
```

And if the path is not wrapped into an array, this is the CMake failure:

```
openal-soft| Found CMake: C:\Program Files\CMake\bin\cmake.EXE (3.26.3)

| Configuring the build directory with CMake version 3.26.3
| Running CMake with: -G Ninja -DCMAKE_INSTALL_PREFIX=c:\ -DCMAKE_BUILD_TYPE=RelWithDebInfo -DALSOFT_UTILS=False -DALSOFT_NO_CONFIG_UTIL=True -DALSOFT_EXAMPLES=False -DALSOFT_INSTALL_CONFIG=False -DALSOFT_INSTALL_HRTF_DATA=False -DALSOFT_INSTALL_AMBDEC_PRESETS=False -DALSOFT_BACKEND_ALSA=False -DALSOFT_BACKEND_OSS=False -DALSOFT_BACKEND_SOLARIS=False -DALSOFT_BACKEND_SNDIO=False -DALSOFT_BACKEND_PORTAUDIO=False -DALSOFT_BACKEND_PULSEAUDIO=False -DALSOFT_BACKEND_COREAUDIO=True -DALSOFT_BACKEND_JACK=False -DALSOFT_BACKEND_OPENSL=False -DALSOFT_BACKEND_WAVE=True -DALSOFT_BACKEND_WINMM=False -DALSOFT_BACKEND_DSOUND=True -DALSOFT_BACKEND_WASAPI=False -DALSOFT_CPUEXT_NEON=False -DCMAKE_DISABLE_FIND_PACKAGE_WindowsSDK=True -DCMAKE_DISABLE_FIND_PACKAGE_Git=True -DCMAKE_OSX_DEPLOYMENT_TARGET=  
|   - build directory:          C:/Users/Amalia/Desktop/Rachel/libAudio/build/deps/openal-soft/__CMake_build
|   - source directory:         C:/Users/Amalia/Desktop/Rachel/libAudio/deps/openal-soft
|   - toolchain file:           C:/Users/Amalia/Desktop/Rachel/libAudio/build/deps/openal-soft/__CMake_build/CMakeMesonToolchainFile.cmake
|   - preload file:             C:/Users/Amalia/AppData/Roaming/Python/Python310/site-packages/mesonbuild/cmake/data/preload.cmake
|   - trace args:               --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt
|   - disabled policy warnings: [CMP0025, CMP0047, CMP0056, CMP0060, CMP0065, CMP0066, CMP0067, CMP0082, CMP0089, CMP0102]

| Running with expanded trace output on.
| Not searching for unused variables given on the command line.
| Trace will be written to cmake_trace.txt
| -- The C compiler identification is unknown
| -- The CXX compiler identification is unknown
| CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.26/Modules/CMakeFindBinUtils.cmake:220 (find_program):
| Policy CMP0126 is not set: set(CACHE) does not remove a normal variable of
| the same name.  Run "cmake --help-policy CMP0126" for policy details.  Use
| the cmake_policy command to set the policy and suppress this warning.

| For compatibility with older versions of CMake, normal variable "CMAKE_AR"
| will be removed from the current scope.
| Call Stack (most recent call first):
| C:/Program Files/CMake/share/cmake-3.26/Modules/CMakeDetermineCXXCompiler.cmake:202 (include)
| CMakeLists.txt:56 (project)
| This warning is for project developers.  Use -Wno-dev to suppress it.

| CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.26/Modules/CMakeFindBinUtils.cmake:220 (find_program):
| Policy CMP0126 is not set: set(CACHE) does not remove a normal variable of
| the same name.  Run "cmake --help-policy CMP0126" for policy details.  Use
| the cmake_policy command to set the policy and suppress this warning.

| For compatibility with older versions of CMake, normal variable
| "CMAKE_LINKER" will be removed from the current scope.
| Call Stack (most recent call first):
| C:/Program Files/CMake/share/cmake-3.26/Modules/CMakeDetermineCXXCompiler.cmake:202 (include)
| CMakeLists.txt:56 (project)
| This warning is for project developers.  Use -Wno-dev to suppress it.

| CMake Error at CMakeLists.txt:56 (project):
| The CMAKE_C_COMPILER:

| Files/Microsoft

| is not a full path and was not found in the PATH.

| Tell CMake where to find the compiler by setting either the environment
| variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
| the compiler, or to the compiler name if it is in the PATH.


| CMake Error at CMakeLists.txt:56 (project):
| The CMAKE_CXX_COMPILER:

| Files/Microsoft

| is not a full path and was not found in the PATH.

| Tell CMake where to find the compiler by setting either the environment
| variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
| to the compiler, or to the compiler name if it is in the PATH.


| -- Configuring incomplete, errors occurred!
```

This was a great catch, I've just run over the source and fixed where these paths were being forcefully split.